build-system/tasks/dep-check.js

```diff
@@ -311,7 +310,14 @@ async function depCheck() {
     .then(runRules)
     .then(errorsFound => {
       if (errorsFound) {
-        process.exit(1);
+        log(
+          yellow('NOTE:'),
+          'If a dependency is valid, add it to one of the whitelists in',
+          cyan('build-system/dep-check-config.js')
+        );
+        const reason = new Error('Dependency checks failed');
+        reason.showStack = false;
+        return Promise.reject(reason);
```

Comment:
does `handlerProcess` end on its own?

Reply:
Yes, its lifetime is tied to the parent `gulp` process.